### PR TITLE
[fix] updated the config storage path for linux

### PR DESCRIPTION
**Pull request by poshul**
# Description
addresses https://github.com/OpenMS/OpenMS/issues/3681

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Can you add that to the changelog before we forget?

**Comment by poshul** [comment]
rebuild jenkins

**Comment by poshul** [comment]
The Win 10 test failure looks like a VM issue, and not specific to this PR

File: CHANGELOG [hunk 1]
```diff
@@ -92,6 +92,7 @@ Various:
 - Some further clarifications, standardization/consolidation of outputs formats/consistency
 - Continued fixes/improvement of documentation
 - Removed deprecated version of PeakPickerHiRes (LowMemPeakPickerHiResRandomAccess and LowMemPeakPickerHiRes). Their functionality can be accessed through options of PeakPickerHiRes
+- moved the config storage location for unix installations to comply with XDG guidelines.
```
Comment (cbielow) [inline CHANGELOG:95]: can we spell out `XDG` here, because many people might not know it

File: src/openms/source/SYSTEM/UpdateCheck.cpp [hunk 2]
```diff
@@ -73,26 +73,33 @@ namespace OpenMS
     platform = "Win";
 #elif __APPLE__
     platform = "Mac";
-#else
+#elif __unix__
```
Comment (cbielow) [inline src/openms/source/SYSTEM/UpdateCheck.cpp:76]: not sure about that part.
XDG mentions KDE, GNOME, XServer and alike, and this sounds like Linux to me (`__linux__`).
However, `__unix__` seems to be defined on our debian as well, which I find very suprising (linux != unix, right? :-) )

File: src/openms/source/SYSTEM/File.cpp [hunk 3]
```diff
@@ -621,8 +621,12 @@ namespace OpenMS
   Param File::getSystemParameters()
   {
     String home_path = File::getOpenMSHomePath();
-
+    //Comply with https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html on unix identifying systems
+    #ifdef __unix__
+    String filename = home_path + "/.config/OpenMS/OpenMS.ini";
```
Comment (cbielow) [inline src/openms/source/SYSTEM/File.cpp:626]: technically, you'd need to query `$XDG_CONFIG_HOME` first and use that as a full prefix. 
I'm not sure though if its usually defined or if the fallback is used in most cases anyway....

From the docs:
```
$XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used. 
```

File: src/openms/source/SYSTEM/UpdateCheck.cpp (hunk 2 above)
Comment (poshul) [reply src/openms/source/SYSTEM/UpdateCheck.cpp]: I've added a check for __linux__ first and then __unix__ if its not linux. According to http://web.archive.org/web/20191012035921/http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system all linux variants have __unix__ set.

File: src/openms/source/SYSTEM/UpdateCheck.cpp [hunk 4]
```diff
@@ -73,26 +73,38 @@ namespace OpenMS
...
 
     // write to tmp + userid folder
 
     // e.g.: OpenMS_Default_Win_64_FeatureFinderCentroided_2.0.0
     String tool_version_string;
+    String config_path;
+    //Comply with https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html on unix identifying systems
+    #ifdef __unix__
+    if(getenv("XDG_CONFIG_HOME"))
+      config_path = String(getenv("XDG_CONFIG_HOME")) + "/.config/OpenMS";
```
Comment (cbielow) [inline src/openms/source/SYSTEM/UpdateCheck.cpp:92]: from how I read the docs, `$XDG_CONFIG_HOME` already includes `..config/`, or am I misreading that?

File: src/openms/source/SYSTEM/File.cpp (hunk 3 above)
Comment (cbielow) [inline src/openms/source/SYSTEM/File.cpp:626]: should this also check for XDG_CONFIG_HOME ?

File: src/openms/source/SYSTEM/UpdateCheck.cpp (hunk 4 above)
Comment (poshul) [reply src/openms/source/SYSTEM/UpdateCheck.cpp:92]: Reading the doc again, I believe you are correct. I'll change that.

**Review by cbielow — APPROVED** [review]
thanks! Looks awesome :)

**Comment by timosachsenberg** [comment]
is this also relevant for https://github.com/OpenMS/OpenMS/blob/c2555c3278de34a5f261df9a0915f42a8bfada3c/src/openms_gui/source/VISUAL/APPLICATIONS/GUITOOLS/TOPPView.cpp#L94 ?

**Pull request merged**